### PR TITLE
TST: tighten xfails

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/indexes/test_numpy_compat.py [hunk 1]
```diff
@@ -74,14 +74,16 @@ def test_numpy_ufuncs_basic(index, func):
...
-def test_numpy_ufuncs_other(index, func):
+def test_numpy_ufuncs_other(index, func, request):
     # test ufuncs of numpy, see:
     # https://numpy.org/doc/stable/reference/ufuncs.html
 
     if isinstance(index, (DatetimeIndex, TimedeltaIndex)):
         if isinstance(index, DatetimeIndex) and index.tz is not None:
             if func in [np.isfinite, np.isnan, np.isinf]:
-                pytest.xfail(reason="__array_ufunc__ is not defined")
+                if not np_version_under1p17:
```
Comment (jreback) [inline pandas/tests/indexes/test_numpy_compat.py:84]: why do we not prefer an inside use of `pytest.xfail` and this format?

Comment (jbrockmendel) [reply pandas/tests/indexes/test_numpy_compat.py:84]: using pytest.xfail breaks out immediately, so you dont get alerted if the xfail has been fixed.

Comment (jreback) [reply pandas/tests/indexes/test_numpy_compat.py:84]: interesting; it doesn't react to strict=True?

Comment (jbrockmendel) [reply pandas/tests/indexes/test_numpy_compat.py:84]: AFAIK thats only when used as a decorator `@pytest.mark.xfail(strict=foo)`

Comment (jreback) [reply pandas/tests/indexes/test_numpy_compat.py:84]: kk

can u update the way to write tests in the docs (open an issue or PR)

also prob need to have a rule which prohibits (open an issue)

**Pull request merged**